### ISSUE TITLE
`lines-around-comment`: `allowTypeStart`

### ISSUE THREAD

**Issue by DetachHead** [eslint-stylistic/eslint-stylistic#1245]
### Clear and concise description of the problem

`allowInterfaceStart` only works on interfaces, but not other types:
```ts
type Foo = {
  /** error here */
  foo: {
    /** error here */
    bar: number;
  };
};
```

### Suggested solution

add a new option: `allowTypeStart` which should work on all object types

### Alternative

rename & repurpose `allowInterfaceStart`

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by morgan-coded** [comment]
Use `allowTypeStart: true`; the option is already present in the rule:

```js
"@stylistic/lines-around-comment": ["error", {
  "beforeBlockComment": true,
  "allowTypeStart": true,
}]
```

The rule maps `Type` to `TSTypeLiteral`, so this applies to both the outer `type Foo = { ... }` and the nested `foo: { ... }` type literal in your example. The TypeScript coverage includes both line and block comments.

**Comment by DetachHead** [comment]
huh, i swear i tried that and it didn't work, but i tried it again and it works now. thanks
